### PR TITLE
iay: init at 0.4.0

### PR DESCRIPTION
**Pull request by aaqaishtyaq**
This PR aims to add [iay](https://github.com/aaqaishtyaq/iay), a shell prompt for bash and zsh. 

Build the package using `nix-build -A iay` on darwin and linux. 

Signed-off-by: Aaqa Ishtyaq <aaqaishtyaq@gmail.com>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Have you ever read the docs?

**Comment by aaqaishtyaq** [comment]
> Have you ever read the docs?
@AndersonTorres What exactly are you talking about?

**Comment by aaqaishtyaq** [comment]
@AndersonTorres Please review.

File: pkgs/tools/misc/iay/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0r2yp34gxkh32amrysfj1jg543dh0kyqxzcx0zyi6a8y9232d8ky";
+  };
+
+  cargoHash = "sha256-SMqiwM6LrXXjV4Mb2BY9WbeKKPkxiYxPyZ4aepVIAqU=";
+
+  meta = with lib; {
+    description =
+      "iay! The minimalistic, blazing-fast, and extendable prompt for bash and zsh.";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/iay/default.nix:38]:
```suggestion
    description = "Minimalistic, blazing-fast, and extendable prompt for bash and zsh";
```

short descriptions should never be self-referent and should never end with punctuation marks

File: pkgs/tools/misc/iay/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    homepage = "https://github.com/aaqaishtyaq/iay";
+    license = licenses.mit;
+    platforms = platforms.all;
```
Comment (AndersonTorres) [inline pkgs/tools/misc/iay/default.nix:41]:
```suggestion
```

Rust is not supported on all platforms.

File: pkgs/tools/misc/iay/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+, Security
+, Foundation
+, Cocoa
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "iay";
+  version = "0.4.0";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
```
Comment (AndersonTorres) [inline pkgs/tools/misc/iay/default.nix:17]: `src` should be here, instead of `buildInputs`

File: pkgs/tools/misc/iay/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    Security
+    Foundation
+    Cocoa
+  ]
+  ++ [ openssl pkg-config ]
```
Comment (AndersonTorres) [inline pkgs/tools/misc/iay/default.nix:22]: 1. pkg-config goes on `nativeBuildInputs`
2. the list of optionals should be last on this concatenation.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -186,6 +186,12 @@
     githubId = 7755101;
     name = "Aaron Andersen";
   };
+  aaqaishtyaq = {
```
[the remark this answers is not in the document]
Comment (aaqaishtyaq) [reply maintainers/maintainer-list.nix:189]: sure

File: pkgs/tools/misc/iay/default.nix (hunk 1 above)
Comment (aaqaishtyaq) [reply pkgs/tools/misc/iay/default.nix]: Done. Thank you.

File: pkgs/tools/misc/iay/default.nix (hunk 2 above)
Comment (aaqaishtyaq) [reply pkgs/tools/misc/iay/default.nix]: Got it. Added platforms for which I have done testing.

File: pkgs/tools/misc/iay/default.nix (hunk 3 above)
Comment (aaqaishtyaq) [reply pkgs/tools/misc/iay/default.nix]: Done

File: pkgs/tools/misc/iay/default.nix (hunk 4 above)
Comment (aaqaishtyaq) [reply pkgs/tools/misc/iay/default.nix]: Sure.

**Comment by aaqaishtyaq** [comment]
@AndersonTorres Please re-review.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -14624,6 +14624,10 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Security;
   };
 
+  iay = callPackage ../tools/misc/iay {
+    inherit (darwin.apple_sdk.frameworks) Security Foundation Cocoa;
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:14628]:
```suggestion
    inherit (darwin.apple_sdk.frameworks) AppKit Security Foundation Cocoa;
```

File: pkgs/tools/misc/iay/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  NIX_LDFLAGS = lib.optionals stdenv.isDarwin [ "-framework" "AppKit" ];
+
+  meta = with lib; {
+    description =
+      "Minimalistic, blazing-fast, and extendable prompt for bash and zsh";
+    homepage = "https://github.com/aaqaishtyaq/iay";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin"];
```
Comment (AndersonTorres) [inline pkgs/tools/misc/iay/default.nix:45]:
```suggestion
    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/iay/default.nix [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  NIX_LDFLAGS = lib.optionals stdenv.isDarwin [ "-framework" "AppKit" ];
+
+  meta = with lib; {
+    description =
+      "Minimalistic, blazing-fast, and extendable prompt for bash and zsh";
+    homepage = "https://github.com/aaqaishtyaq/iay";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" ];
```
Comment (AndersonTorres) [inline pkgs/tools/misc/iay/default.nix:45]: Are you testing this package locally on these specific platforms? If not, just remove this line, its value is implicitly set by Nixpkgs.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
> Hey @AndersonTorres, `passthru.tests` for `aarch64-darwin` is stuck in the queue.

The ofborg infrastructure for Arm-Darwin is a bit problematic. Just ignore this for now.

File: pkgs/tools/misc/iay/default.nix (hunk 8 above)
Comment (aaqaishtyaq) [reply pkgs/tools/misc/iay/default.nix:45]: Sure, I will remove this line. I could only test it on these 3 specific platforms/arch.

Comment (aaqaishtyaq) [reply pkgs/tools/misc/iay/default.nix]: I have removed `platforms`. Please re-review.

**Comment by aaqaishtyaq** [comment]
>The ofborg infrastructure for Arm-Darwin is a bit problematic. It becomes stuck even on easy packages.
Just ignore this for now.

Sure.

**Comment by AndersonTorres** [comment]
> This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:
> 
> https://discourse.nixos.org/t/prs-ready-for-review/3032/1579

Why?

**Comment by aaqaishtyaq** [comment]
>Why?

Sorry, I thought we have to post in discourse for PR review. Apologies. 

Thank you for merging the PR.

**Comment by AndersonTorres** [comment]
Implicitly closes https://github.com/NixOS/nixpkgs/pull/204936

**Pull request merged**